doc/source/example-newsreader.rst

```diff
@@ -1,29 +1,96 @@
 News Reader
 ===========
 
-This document shows how to implement a simple news reader using Ray. The reader
-consists of a simple Vue.js `frontend`_ and a backend consisting of a Flask
-server and a Ray actor. View the `code for this example`_.
+This document walks you through implementing a simple news reader using Ray.
+The reader consists of a simple Vue.js frontend and a Ray backend.
+It has a simple recommendation engine built in, which you will train on upvotes
+from hackernews and which learns online as the user stars articles.
+
+It looks like this (the frontend is based on `QReader`_), here the recommended
+articles are in yellow:
+
+.. image:: newsreader.jpg
+
+View the `code for this example`_.
 
 To run this example, you will need to install NPM and a few python dependencies.
 
 .. code-block:: bash
 
   pip install atoma
   pip install flask
+  pip install vowpalwabbit
+
 
+Training the model
+------------------
+
+To train the model on the Hackernews dataset, run the following in the
+``examples/newsreader`` folder:
+
+.. code-block:: bash
 
-To use this example you need to
+  wget https://archive.org/download/14566367HackerNewsCommentsAndStoriesArchivedByGreyPanthersHacker/14m_hn_comments_sorted.json.bz2
+  bunzip2 14m_hn_comments_sorted.json.bz2
+
+Open a python interpreter and enter the following:
+
+.. code-block:: python
+
+  import pandas as pd
+  import training
+
+  records = training.load_hn_submissions("14m_hn_comments_sorted.json")
+  df = pd.DataFrame(records, columns=["title", "score"])
+  df.sort_values(by="score", ascending=False)
```

Comment:
Instead of making people copy and paste all of this code from the readme, why not just make it part of the application? E.g., it can be done in `server.py` and the model can be passed directly into the server.